<a href="https://colab.research.google.com/github/sclaudiobr/algoritmos-py/blob/master/Ajustamento01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

As observações conduzidas pelo ser humano possuem características comuns pela presença de erros de mensuração. Erros que decorrem não apenas de falhas humanas, mas também da imperfeição do equipamento e da influência das condições ambientais nas quais se processa a mensuração. Para o Prof. Camil Gemael: “Quem dá os primeiros passos na análise de observações começa por fazer uma concessão: abdicar da pretensão de obter o verdadeiro valor de uma grandeza medida”[[¹](https://topografia.paginas.ufsc.br/files/2014/09/Ajustamento-Cap2.pdf)]. A incerteza no resultado de uma mensuração isolada, nos leva naturalmente à repetição das observações. Gerando assim uma pluralidade de observações, das quais sabemos que são incorretas por conta das imperfeições.  Quando tratamos de observações os modelos matemáticos são fundamentais para explicitar os resultados que procuramos. Os modelos matemático Funcional e Modelo Estocástico são definidos para tratar os dados físicos e estatísticos de observações, respectivamente. Onde O modelo funcional é composto por relações que descrevem a geometria ou características físicas do problema em questão,  tal como a determinação da área de um terreno retangular através da multiplicação dos seus lados a e b, cujo modelo funcional é A = ab. O modelo estocástico por sua vez, é composto pelo conjunto de relações que descrevem as propriedades estatísticas dos elementos envolvidos no modelo funcional. Diante deste contexto o ajustamento de observações proporciona o resultado bem como estima a precisão da solução adotada,  extraindo um resultado que seja único e que possa representar com confiança a grandeza medida [[²](https://www.scielo.br/scielo.php?pid=S1982-21702011000200007&script=sci_arttext)]. O Método dos Mínimos Quadrados (MMQ) é um dos critérios mais utilizados para o ajustamento de observações geodésicas. Este método adota uma solução única através de equações que minimizam a soma do quadrado dos erros aleatórios, e estimar a precisão de tais incógnitas e a eventual correlação entre elas. Os passos básicos para a realização dos ajustamentos serão descritos neste documento, os quais constam prmeiramento da obtenção dos parâmetros estocásticos dos dados e posterior aplicação do modelo matemático com os parãmetros encontrados.

¹Claudio R. O da Silva

# Aplicação Prática

## Foram levantados no Campus da UFSM um terreno de forma retangular, com estacas nos 4 cantos formando os vértices A,B,C e D, sobre o mesmo foram realizadas várias medidas apresentadas na tabela:

<p>&nbsp;</p>
<table style="width: 448px;">
<tbody>
<tr>
<td style="width: 79px;">
<p><strong>i</strong></p>
</td>
<td style="width: 156px;">
<p><span style="font-weight: 400;">Valores </span><span style="font-weight: 400;">a</span><span style="font-weight: 400;">i</span><span style="font-weight: 400;"> (m)</span></p>
</td>
<td style="width: 197px;">
<p><span style="font-weight: 400;">Valores </span><span style="font-weight: 400;">b</span><span style="font-weight: 400;">i</span><span style="font-weight: 400;"> (m)</span></p>
</td>
</tr>
<tr>
<td style="width: 79px;">
<p><span style="font-weight: 400;">1</span></p>
</td>
<td style="width: 156px;">
<p><span style="font-weight: 400;">80,10</span></p>
</td>
<td style="width: 197px;">
<p><span style="font-weight: 400;">20,16</span></p>
</td>
</tr>
<tr>
<td style="width: 79px;">
<p><span style="font-weight: 400;">2</span></p>
</td>
<td style="width: 156px;">
<p><span style="font-weight: 400;">80,30</span></p>
</td>
<td style="width: 197px;">
<p><span style="font-weight: 400;">20,24</span></p>
</td>
</tr>
<tr>
<td style="width: 79px;">
<p><span style="font-weight: 400;">3</span></p>
</td>
<td style="width: 156px;">
<p><span style="font-weight: 400;">79,90</span></p>
</td>
<td style="width: 197px;">
<p><span style="font-weight: 400;">20,40</span></p>
</td>
</tr>
<tr>
<td style="width: 79px;">
<p><span style="font-weight: 400;">4</span></p>
</td>
<td style="width: 156px;">
<p><span style="font-weight: 400;">80,20</span></p>
</td>
<td style="width: 197px;">
<p><span style="font-weight: 400;">19,80</span></p>
</td>
</tr>
<tr>
<td style="width: 79px;">
<p><span style="font-weight: 400;">5</span></p>
</td>
<td style="width: 156px;">
<p><span style="font-weight: 400;">79,90</span></p>
</td>
<td style="width: 197px;">
<p><span style="font-weight: 400;">19,60</span></p>
</td>
</tr>
</tbody>
</table>

In [1]:
import numpy as np

In [ ]:
# Nossos dados <- Rode este para o trabalho
a = ([[80.10],[80.30],[79.90],[80.20],[79.90]])
b = ([[20.16],[20.24],[20.40],[19.80],[19.60]])

In [ ]:
# Exemplo <- Rode este para comparar com o exercício fornecido pelo José
#a = ([[40.05],[40.15],[39.95],[40.10],[39.90]])
#b = ([[10.08],[10.12],[10.20],[9.90],[9.80]])

# Média
Medida é uma medida de tendência central, que busca indicar o valor no centro da distribuição de frequências, onde há maior ocorência dos eventos.

# <P>$\bar{a} = \frac{1}{n} \sum_{i=1}^n $
# <p> $\bar{b} = \frac{1}{n} \sum_{i=1}^n $

In [ ]:
# media ('a)
ma = np.mean(a)
mb = np.mean(b)
print('Média de a: ',ma)
print('Média de b: ',mb)

Média de a:  80.08
Média de b:  20.04


In [ ]:
# Distancia entre ai e o valor da média ('a)
# ai - 'a
a - np.mean(a)

array([[ 0.02],
       [ 0.22],
       [-0.18],
       [ 0.12],
       [-0.18]])

In [ ]:
sum(a-ma)

array([1.42108547e-14])

In [ ]:
# bi - 'b
b - np.mean(b)

array([[ 0.12],
       [ 0.2 ],
       [ 0.36],
       [-0.24],
       [-0.44]])

In [ ]:
# (ai - 'a)^2
(a - np.mean(a))**2

array([[0.0004],
       [0.0484],
       [0.0324],
       [0.0144],
       [0.0324]])

In [ ]:
# (bi - 'b)^2
(b - np.mean(b))**2

array([[0.0144],
       [0.04  ],
       [0.1296],
       [0.0576],
       [0.1936]])

# Variância
É uma medida de dispersão estatística, indicando "o quão longe" em geral os  valores se encontram-se da média. Representando a variabilidade do evento em torno da média, calculada pela média de $(a_i - \bar a)$. O cálculo é efetuado para cada variável isoladamente, e os resultados indicam quanto maior  é, mais os valores estão distantes da média, ou maior a variabilidade dos dados, e quanto mais próximos da média, menor o valo e consequantimente menor a variabilidade.
<p>
Variância de a: $ \;\;\sigma_a^2 = \frac {1}{n-1} \sum_{1=1}^n (a_i-\bar{a})^2$

In [ ]:
# Variância de a
va = np.sum((a - np.mean(a))**2)/(len(a)-1)
print('Variância de a: ', va)

Variância de a:  0.03199999999999878


Variância de b:
$\;\; \sigma_b^2 = \frac {1}{n-1} \sum_{1=1}^n (a_i-\bar{b})^2$

In [ ]:
# Variância de b
vb = np.sum((b - np.mean(b))**2)/(len(b)-1)
print('Variância de b: ',vb)

Variância de b:  0.1087999999999992


# Covariância
<p>A covariância é por vezes chamada de medida de dependência linear entre as duas variáveis aleatórias. Medindo qual o grau de dependência que possuem tem entre si em uma escala de valores de 0 a 1. Se o valor de covariância for igual a zero, indica que as variáveis são independentes, ou não possuem correlação alguma. Se o valor se aproxima de um, indica que as variáveis possuem padrões que acorrem simultaneamente, ou seja, possuel alto grau de correlação. Esta métrica é aplicada na formulação de hipóteses afirmativas ou negativas sobre o comportamebto de duas variáveis.
<p>
# $cov(x,y) = \frac{\sum (x_i - x) \cdot (yi-y)}{n-1} $

In [ ]:
# Covariância de (ai - 'a)*(bi - 'b)
(a - np.mean(a))*(b - np.mean(b))

array([[ 0.0024],
       [ 0.044 ],
       [-0.0648],
       [-0.0288],
       [ 0.0792]])

In [ ]:
# Soma de Covariância ab
cvab = sum((a - np.mean(a))*(b - np.mean(b)))/(len(a)-1)
print('Covariância de ab:',cvab)
ea = np.ndarray.item(cvab)

Covariância de ab: [0.008]


In [ ]:
# Soma de Covariância ba
cvba = sum((a - np.mean(a))*(b - np.mean(b)))/(len(a)-1)
print('Covariância de ab:',cvba)
eb = np.ndarray.item(cvba)

Covariância de ab: [0.008]


1) Matriz de covariância de X
Estimativa da MVC associada ao vetor aleatório :

$\widehat{\sum_{\widehat{x}}} = \begin{bmatrix}
\widehat{\sigma}_{a} & \widehat{\sigma}_{ab} \\ 
\widehat{\sigma}_{ba} & \widehat{\sigma}_{b} 
\end {bmatrix}_{2x2 \; m^2}$

In [ ]:
# Matriz de variância e covariância 2x2 (MVC) de X
mvcx = np.array([[va,ea],[eb,vb]])
print(mvcx)

[[0.032  0.008 ]
 [0.008  0.1088]]


2) Verificação se $\widehat{\sum_{}}_\widehat{x} = \widehat{\sum_{}}_\widehat{x}^i$ é verdadeira través dos autovalores:
<p>Os elementos da diagonal são números reais positivos, porque eles são as variâncias. Matriz positiva definida, porque os “p” autovalores são números reais positivos?
<p>Det. $\widehat{\sum}_\widehat{x}-\widehat{\lambda}_i I = 0$
<p>
MATRIZ DE AUTOVALORES
<P>
$\widehat{\sum_{\widehat{x}}}  - \widehat{\lambda} \cdot I = $
$\begin{bmatrix}
\widehat{\sigma}_{a} & \widehat{\sigma}_{ab} \\ 
\widehat{\sigma}_{ba} & \widehat{\sigma}_{b} 
\end {bmatrix}_{2x2 m^2}$ -
$\begin{bmatrix}\widehat{\lambda}_i & 0 \\ 0 & \widehat{\lambda}_i \end{bmatrix}_{2x2}$
 = 
$\begin{bmatrix}
\widehat{\sigma}_{a} - \widehat{\lambda}_i  & 
\widehat{\sigma}_{ab} \\ 
\widehat{\sigma}_{ba} & \widehat{\sigma}_{b} - \widehat{\lambda}_i 
\end {bmatrix}_{2x2 \; m^2}$
<p>
Seguindo na forma algébrica:

<p>
$Det \widehat{\lambda}_i = [(\widehat{\sigma}_{a} - \widehat{\lambda}_i)\cdot(\widehat{\sigma}_{b} - \widehat{\lambda}_i)]-[\widehat{\sigma}_{ab} \cdot \widehat{\sigma}_{ba}]$<p>

$\widehat{\lambda}_i^2+[(\widehat{\sigma}_{a} \widehat{\lambda}_i-\widehat{\sigma}_{a}\widehat{\lambda}_i)]  - [\widehat{\sigma}_{ab} \cdot \widehat{\sigma}_{ba}] = 0$<p>

In [ ]:
from scipy.linalg import eig
eigenValx,eigenVecx=eig(mvcx)
print("DX = ",((-vb-va)**2)-4*(va*vb-(ea*eb)))
print ("X' = ",eigenValx[0])
print ("X''=",eigenValx[1])

DX =  0.006154240000000023
X' =  (0.03117551784917888+0j)
X''= (0.1096244821508191+0j)


A matriz satisfaz as três condições básicas concomitantemente, portanto é uma MVC.

# 3) Sendo a $\widehat{\sum}_{\widehat{x}}$  verdadeira, estime a MDP associada ao vetor aleatório $\vec{x}$:
<p>
Onde: $\vec{x} = \begin{bmatrix}
a
\\ 
b
\end{bmatrix}_{2x1}$ a MDP associada ao vetor aleatório $\widehat{D}_{\vec{x}}=
\begin{bmatrix}
\sigma_a & 0 \\ 
0 & \sigma_b
\end{bmatrix}_{2x2(Diagonal)}$
<p>
Então:
<p>
$\widehat{D}_{\vec{x}}=
\begin{bmatrix}
\sqrt{\widehat{\sigma}_a^2} & 0 \\ 
0 & \sqrt{\widehat{\sigma}_b^2}
\end{bmatrix}_{2x2(Diagonal)}$
$\widehat{D}_{\widehat{x}} = \begin{bmatrix}
\sqrt{0,032} & 0 \\ 
0 & \sqrt{0,1088} 
\end {bmatrix}_{2x2} \; m^2 $

In [ ]:
print(np.array([[va,ea],[eb,vb]]))

[[0.032  0.008 ]
 [0.008  0.1088]]


In [ ]:
# MDPX
mdpx = np.array([[np.sqrt(va),0],[0,np.sqrt(vb)]])
print(mdpx)

[[0.17888544 0.        ]
 [0.         0.32984845]]


In [ ]:
detDx = np.linalg.det(mdpx)
print(detDx)

0.059005084526673156


## 4) Sendo a $\widehat{\sum}_{\widehat{x}}$   verdadeira, estime a MC associada ao vetor aleatório $\vec{x}$ :

<p>
$\vec{x} = \begin{bmatrix}
a
\\ 
b
\end{bmatrix}_{2x1}$ a MC associada ao vetor aleatório $\widehat{D}_{\vec{x}}=
\begin{bmatrix}
\widehat{r}_{a,a} & \widehat{r}_{a,b} \\ 
\widehat{r}_{b,a} & \widehat{r}_{b,b}
\end{bmatrix}_{2x2}$
<p>
Onde:<p> $\widehat{R}_{\vec{x}}=(\widehat{D}_{\vec{x}})^{-1}\widehat{\sum}_{\vec{x}}(\widehat{D}_{\vec{x}})^{-1}$
<p>
Representação matricial
<p>
$\widehat{R}_{\vec{x}}=
\begin{bmatrix}
1 & \frac{\widehat{\sigma}_{a,b}}{(\sigma_a)(\sigma_b)} \\ 
\frac{\widehat{\sigma}_{b,a}}{(\widehat{\sigma}_{b})(\widehat{\sigma}_{a})} & 1
\end{bmatrix}_{2x2 \;\; (métrica)}$
<p>
$\widehat{R}_{\vec{x}}=
\begin{bmatrix}
1 & \frac{0,032}{(0,17888544)(0,32984845)} \\ 
\frac{0,032}{(0,32984845)(0,17888544)} & 1
\end{bmatrix}_{2x2 \;\; (métrica)}$

In [ ]:
# Matriz de covariância de x
mcx = np.array([[1,ea/(np.sqrt(va)*np.sqrt(vb))],[eb/(np.sqrt(va)*np.sqrt(vb)),1]])
print(mcx,"m2")

[[1.         0.13558154]
 [0.13558154 1.        ]] m2


In [ ]:
detRx = np.linalg.det(mcx)
print('determinande de X:',detRx)

determinande de X: 0.9816176470588257


# 2ª Questão
<p>
Sendo o vetor aleatório $\vec{y}= \begin{bmatrix}P\\ S \end{bmatrix}_{2x1}$  em que “P” é o perímetro e “S” a área do lote

## 1) Estimativa da MVC associada ao vetor aleatório $\vec{y}$ :
<p>
 1ª parte: Modelo Matemático <p>
$\vec{y}= \begin{bmatrix}P\\ S \end{bmatrix}_{2x1}$; $\vec{x}= \begin{bmatrix}P\\ S \end{bmatrix}_{2x1}$
<p> Onde: <p>
$P = f1(a,b) = 2a + 2b = 2(a+b)$<p>
$S = 2(a,b) = a \cdot b $<p>

2ª parte: MVC associada ao vetor aleatório $\vec{x}$:

$\vec{x}= \begin{bmatrix}a\\ b \end{bmatrix}_{2x1}$
$\widehat{\sum}_{\widehat{x}} = \begin{bmatrix}
\widehat{\sigma}_{a}^2 & \widehat{\sigma}_{ab}^2 \\ 
\widehat{\sigma}_{ba}^2 & \widehat{\sigma}_{b}^2 
\end {bmatrix}_{2x2 \; m^2}$

In [ ]:
print(np.array([[va,ea],[eb,vb]]))

[[0.032  0.008 ]
 [0.008  0.1088]]


3ª parte: Lei da propagação das covariâncias<p>
$\widehat{\sum}_{\widehat{x}}$ as covariâncias se propagam em $\widehat{\sum}_{\widehat{y}}$<p>
$\widehat{\sum}_{\widehat{x}} = D \widehat{\sum}_{\widehat{x}} D^T$<p>

$\widehat{\sum}_{\vec{y}} = \begin{bmatrix}
\widehat{\sigma}_{P}^2 & \widehat{\sigma}_{P,S}^2 \\ 
\widehat{\sigma}_{S,P}^2 & \widehat{\sigma}_{S}^2 
\end {bmatrix}_{2x2 \; m^2}$
$ D = 
\begin{bmatrix}
\frac{\partial \vec{y}_i}{\partial \vec{x}}
\end{bmatrix}_{nxp}
$<p>
$
D = \begin{bmatrix}
\frac{\partial P}{\partial a} & \frac{\partial P}{\partial b} \\ 
 \frac{\partial S}{\partial a}& \frac{\partial S}{\partial b} 
\end{bmatrix}_{2x2 ;\ métrica}$<p>
$\frac{\partial P}{\partial a} = \frac{\partial (2a+2b)}{\partial a} = \frac{2 \partial a}{\partial a} +  \frac{2 \partial b}{\partial a}  = 2 \cdot 1 + 0 = 2$<p>
$\frac{\partial P}{\partial b} = \frac{\partial (2a+2b)}{\partial b} = \frac{2 \partial a}{\partial b} +  \frac{2 \partial b}{\partial b}  = 0 + 2 \cdot 1 = 2$<p>
$\frac{\partial S}{\partial a} = \frac{\partial (a \cdot b)}{\partial a} = 1 \cdot b = b$<p>
$\frac{\partial S}{\partial b} = \frac{\partial (a \cdot b)}{\partial b} = 1 \cdot a = a$
<p>
$ D = \begin{bmatrix}
2 & 2 \\ 
 b & a 
\end{bmatrix}_{2x2 ;\ métrica} $


$\widehat{\sum}_\vec{y} = 
\begin{bmatrix}
2 & 2 \\ 
20,04 & 80,08
\end{bmatrix}
\begin{bmatrix}
0,032 & 0,0,008 \\ 
0,008 & 0,1088
\end{bmatrix} \times
\begin{bmatrix}
2 & 20,04 \\ 
2 & 80,08
\end{bmatrix}$
<p>
$\widehat{\sum}_\vec{y} = 
\begin{bmatrix}
0.079999 & 0.233599 \\ 
1.281919 & 8.8730239
\end{bmatrix} \times
\begin{bmatrix}
2 & 20,04 \\ 
2 & 80,08
\end{bmatrix}$
<p>
$
\widehat{\sum}_\vec{y} = 
\begin{bmatrix}
0.627199999 & 20.30988799 \\
20.30988799 &736.24143871
\end{bmatrix}$

In [ ]:
D = (np.array([[2,2],[mb,ma]]))
print(D)

[[ 2.    2.  ]
 [20.04 80.08]]


In [ ]:
print(D.dot(mvcx))

[[0.08     0.2336  ]
 [1.28192  8.873024]]


In [ ]:
print(D.dot(mvcx)[0][0])
print(D.dot(mvcx)[0][1])
print(D.dot(mvcx)[1][0])
print(D.dot(mvcx)[1][1])

0.07999999999999624
0.2335999999999971
1.2819199999999227
8.873023999999923


In [ ]:
Dt = D.transpose()
Dt

array([[ 2.  , 20.04],
       [ 2.  , 80.08]])

In [ ]:
mvcy = ((D.dot(mvcx)).dot(Dt))
print(mvcy)

[[6.27200000e-01 2.03098880e+01]
 [2.03098880e+01 7.36241439e+02]]


In [ ]:
print(mvcy[0][0])
print(mvcy[0][1])
print(mvcy[1][0])
print(mvcy[1][1])

0.6271999999999867
20.30988799999969
20.30988799999969
736.2414387199923


$\widehat{\sum}_{\vec{y}} = \begin{bmatrix}
\widehat{\sigma}_{P}^2 & \widehat{\sigma}_{P,S}^3 \\ 
\widehat{\sigma}_{S,P}^3 & \widehat{\sigma}_{S}^2 
\end {bmatrix}_{2x2 \; m^2}$

<p>
2) Verificação se a matriz $\widehat{\sum}_{\vec{y}}$  é verdadeira:<p>

1º: É uma matriz simétrica, pois $\widehat{\sum}_{\vec{y}} = \widehat{\sum}_{\vec{y}}^t$
<p>
2º:Os elementos da diagonal são números reais positivos, porque eles são as variâncias. 
<p>
3º: Matriz positiva definida, porque os “p” autovalores são números reais positivos?<p>
$\widehat{\sum}_{\vec{y}}-\widehat{\lambda} \cdot I = 0$
<p>
Seguindo na forma algébrica:
<p>
$Det \widehat{\lambda}_i = [(\widehat{\sigma}_{a} - \widehat{\lambda}_i)\cdot(\widehat{\sigma}_{b} - \widehat{\lambda}_i)]-[\widehat{\sigma}_{ab} \cdot \widehat{\sigma}_{ba}]$<p>

$\widehat{\lambda}_i^2+[(\widehat{\sigma}_{a} \widehat{\lambda}_i-\widehat{\sigma}_{a}\widehat{\lambda}_i)]  - [\widehat{\sigma}_{ab} \cdot \widehat{\sigma}_{ba}] = 0$<p>
MATRIZ DE AUTOVALORES
<P>
$\widehat{\sum}_\widehat{y}  - \widehat{\lambda} \cdot I = $
$\begin{bmatrix}
\widehat{\sigma}_{a} & \widehat{\sigma}_{ab} \\ 
\widehat{\sigma}_{ba} & \widehat{\sigma}_{b} 
\end {bmatrix}_{2x2 m^2}$ -
$\begin{bmatrix}\widehat{\lambda}_i & 0 \\ 0 & \widehat{\lambda}_i \end{bmatrix}_{2x2}$
 = 
$\begin{bmatrix}
\widehat{\sigma}_{a} - \widehat{\lambda}_i  & 
\widehat{\sigma}_{ab} \\ 
\widehat{\sigma}_{ba} & \widehat{\sigma}_{b} - \widehat{\lambda}_i 
\end {bmatrix}_{2x2 \; m^2}$
<p>

$\widehat{\sum}_\widehat{y}  - \widehat{\lambda} \cdot I = $
$\begin{bmatrix}
0.627199999 & 20.30988799 \\
20.30988799 &736.24143871 
\end {bmatrix}_{2x2 m^2}$ -
$\begin{bmatrix}\widehat{\lambda}_i & 0 \\ 0 & \widehat{\lambda}_i \end{bmatrix}_{2x2}$
 =
$\begin{bmatrix}
0.62719999 - \widehat{\lambda}_i  & 
20.30988799 \\ 
20.30988799& 736.24143871  - \widehat{\lambda}_i 
\end {bmatrix}_{2x2 \; m^2}$
<p>
$Det \widehat{\lambda}_i = [(0.62719999 - \widehat{\lambda}_i)\cdot(736.2414387  - \widehat{\lambda}_i)]-[ [20.30988799 \cdot 20.30988799]$
<p>
$\widehat{\lambda}_i^2-461,7706226 \widehat{\lambda}_i  - 412,4915502 = 0 x \mapsto \Delta = b^2 - 4ac$

<p>$\Delta = -461,7706226^2 - 4 \cdot (1)- 412,4915502$
<p>$\Delta = 213232 - 1649,966201$



In [ ]:
print(np.array([[va,ea],[eb,vb]]))

[[0.032  0.008 ]
 [0.008  0.1088]]


In [ ]:
eigenValy,eigenVecy=eig(mvcy)
print ("X' = ",eigenValy[0])
print ("X''=",eigenValy[1])

X' =  (0.06688241358131108+0j)
X''= (736.8017563064111+0j)


A matriz satisfaz as três condições básicas concomitantes, portanto é uma MVC

3) Sendo a $\widehat{\sum}_{\vec{y}}$  verdadeira, estime a MDP associada ao vetor aleatório $\widehat{y}$.<p>

$\vec{y}= \begin{bmatrix}P\\ S \end{bmatrix}_{2x1}$ a MDP associada ao vetor aleatório $\widehat{D}_{\widehat{y}} = \begin{bmatrix}
\widehat{\sigma}_{p} & 0 \\ 
0 & \widehat{\sigma}_{S} 
\end {bmatrix}_{2x2 \; m^2}$<p>

$\widehat{D}_{\widehat{y}} = \begin{bmatrix}
\sqrt{\widehat{\sigma}_{p}^2} & 0 \\ 
0 & \sqrt{\widehat{\sigma}_{S}^2} 
\end {bmatrix}_{2x2 \; m^2}$


$\widehat{D}_{\widehat{y}} = \begin{bmatrix}
\sqrt{0,627199999} & 0 \\ 
0 & \sqrt{736.24143871} 
\end {bmatrix}_{2x2 \; m^2}$ <p>

$\widehat{D}_{\widehat{y}} = \begin{bmatrix}
0,79195959 & 0 \\ 
0 & 27,13376934
\end {bmatrix}_{2x2 \; m^2}$

In [ ]:
print(mvcy)

[[6.27200000e-01 2.03098880e+01]
 [2.03098880e+01 7.36241439e+02]]


In [ ]:
dy = np.array([[mvcy[0][0],mvcy[1][1]]])
dys= np.array([[np.sqrt(mvcy[0][0]),0],[0,np.sqrt(mvcy[1][1])]])
print(dy)
print("===="*6)
print(dys)

[[6.27200000e-01 7.36241439e+02]]
[[ 0.79195959  0.        ]
 [ 0.         27.13376934]]


## 4) Sendo a $\widehat{\sum}_{\widehat{y}}$   verdadeira, estime a MC associada ao vetor aleatório $\vec{y}$ :

<p>
$\vec{y} = \begin{bmatrix}
a
\\ 
b
\end{bmatrix}_{2x1}$ a MC associada ao vetor aleatório $\widehat{D}_{\vec{y}}=
\begin{bmatrix}
\widehat{r}_{P,P} & \widehat{r}_{P,S} \\ 
\widehat{r}_{S,P} & \widehat{r}_{S,S}
\end{bmatrix}_{2x2}$
<p>
Onde:<p> $\widehat{R}_{\vec{y}}=(\widehat{D}_{\vec{y}})^{-1}\widehat{\sum}_{\vec{y}}(\widehat{D}_{\vec{y}})^{-1}$
<p>
Representação matricial
<p>
$\widehat{R}_{\vec{y}}=
\begin{bmatrix}
1 & \frac{\widehat{\sigma}_{P,S}}{(\sigma_P)(\sigma_S)} \\ 
\frac{\widehat{\sigma}_{S,P}}{(\widehat{\sigma}_{S})(\widehat{\sigma}_{P})} & 1
\end{bmatrix}_{2x2 \;\; (métrica)}$


$\widehat{R}_{\vec{y}}=
\begin{bmatrix}
1 & \frac{20,30988799}{(0,7837)(27,13376934)} \\ 
\frac{20,30988799}{(27,13376934)(0,7837)} & 1
\end{bmatrix}_{2x2 \;\; (metrica)}$
<p>
$\widehat{R}_{\vec{y}}=
\begin{bmatrix}
1 & 0,94513615 \\ 
0,94513615 & 1
\end{bmatrix}_{2x2 \;\; (metrica)}
$

In [ ]:
inv = np.linalg.inv(dys).dot(mvcy)

In [ ]:
inv.dot(np.linalg.inv(dys))

array([[1.        , 0.94513615],
       [0.94513615, 1.        ]])

##REFERÊNCIAS

<p>
[1] Ara, A. B. <b> Introdução à Estatística </b>. Rio de janrio-RJ: Editora Blucher, 2003. 9788521214915. Disponível em: https://integrada.minhabiblioteca.com.br/#/books/9788521214915/. Acesso em: 15 Sep 2020
<p>[2] Klein, I., Matsuoka, M. T., de Souza, S. F., & Veronez, M. R. (2011). Ajustamento de observações: Uma interpretação geométrica para o método dos mínimos quadrados. Boletim de Ciencias Geodesicas, 17(2), 272–294. https://doi.org/10.1590/s1982-21702011000200007.
]